### PR TITLE
refactor: index.jsx to use getServerSideProps

### PR DESCRIPTION
**Pull request by theogainey**
### Description
refactor index.jsx to use getServerSideProps instead of useEffect
steps taken:
- move data fetching logic from useEffect to getServerSideProps
- remove state management logic `useEffect` & `useState`
- remove tests from `__tests__/index.test.jsx ` that no longer reflect desired application logic 
- add `services/apiService.test.js` to test data fetching logic 

### Spec

See Story: [207]((https://sparkbox.atlassian.net/browse/FSA22V1-207))

### Validation

* [ ] This PR has code changes, and our linters still pass.
* [ ] This PR has new code, so new tests were added or updated, and they pass.
* [ ] This PR has copy changes, so copy was proofread and approved.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
2. Pull down all related branches.
3. Confirm all tests pass.
4. Confirm with react dev tools that Home page(`/`) has `teamMembers` as a prop. Do this step with both in a local dev environment and in the preview deploy

### CODE REVIEW

File: services/apiService.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    afterEach(() => {
+      global.fetch = defaultFetchReference;
+    });
+
+    it('returns team memebers as an array of objects ', async () => {
+      const actual = await apiService.getAllTeamMembers('url');
+      expect(Array.isArray(actual)).toBe(true);
+    });
+
+    it('returns the object key for each team memeber as an id field in the team memebers data object key:{data} => {id:key, data}', async () => {
```
Comment (dustin-jw) [inline services/apiService.test.js:23]: Whoops, couple typos: 'memeber' -> 'member'

File: services/apiService.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import MOCK_DATA from '../mock-data/fellowship.json';
```
Comment (dustin-jw) [inline services/apiService.test.js:1]: Thanks for adding tests for this!

File: services/apiService.test.js (hunk 1 above)
Comment (dustin-jw) [inline services/apiService.test.js:23]: This is a pretty long test description – I'd maybe simplify it to something like 'returns the team member's id as part of the data object'

**Review by dustin-jw — DISMISSED** [review]
Nice work! I have some extremely minor feedback that I won't let block the PR, and I'll let @arnest00 wrap it up from here!

File: services/apiService.test.js (hunk 2 above)
Comment (arnest00) [reply services/apiService.test.js:1]: Great work on this test!

**Review by arnest00 — APPROVED** [review]

**Pull request merged**